x/ibc-hooks/contracts/crosschain-swaps/README.md

```diff
@@ -0,0 +1,214 @@
+# Crosschain Swaps
+
+The following contract is a *swap and forward* contract that takes the received
+tokens, swaps them via a swaprouter contract, and sends them to an IBC receiver.
+
+The goal is to use this contract to provide *crosschain swaps*: sending an ICS20
+transfer on chain A, receiving it on osmosis, swapping for a different token,
+and forwarding to a different chain.
+
+## Instantiation
+
+To instantiate the contract, you need to specify the following parameters:
+
+ * swap_contract: the swaprouter contract to be used
+ * track_ibc_sends: true|false. Specifies if the contract should track the sent ibc packets for recovery. This should be false in v1
+ * channels: a list of (bech32 prefix, channel_id) that the contract will allow. 
+
+### Example instantiation message
+
+``` json
+{"swap_contract": "osmo1thiscontract", "track_ibc_sends": false, "channels": [["cosmos", "channel-0"], ["juno", "channel-42"]]}
+```
+
+## Usage
+
+### Via IBC
+
+Assuming the current implementation of the wasm middleware on Osmosis, the memo
+of an IBC transfer to do crosschain swaps would look as follows:
+
+``` json
+{"wasm": {
+    "contract": "osmo1crosschainswapscontract", 
+    "msg": {
+        "osmosis_swap": {
+            "input_coin": {"denom":"token0","amount":"1000"}, 
+            "output_denom":"token1",
+            "slippage":{"max_slippage_percentage":"5"},
+            "receiver":"juno1receiver",
+            "failed_delivery":null
+            "next_memo":null
+        }
+    }
+}}
+```
+
+Channels are determined by the prefixes specified in the contract during
+instantiation, so the user needs to provide a receiver with the supported
+prefix. In future (once ack/timeout tracking is supported by the chain), we
+could enable support for any channel by specifying it as a parameter.
+
+The `slippage` can be set to a percentage of the twap price (as shown above), or as
+the minimum amount of tokens expected to be received: `{"min_output_amount": "100"}`.
```

Comment:
I think we need some key adjustment for twap, e.g.
```
{
  "slippage": {
     "twap": {"window": "24h", "slippage_percent": "5"},
     "min amount out": {<however you specify amount>},
}
```

if twap window left blank infer some value. Always check min amount out if present

Reply:
Maybe we should also add some docs that slippage is coming from swaprouter?

Reply:
I want to move Slippage (and some other utilities) to a package in the outpost contract repo.  Makes sense to add the window.

Not sure what you mean by: Always check min amount out if present

Reply:
hmm, I guess that would work. But I changed it to an enum, so it can now be either `{"twap":{"slippage_percentage":"20","window_seconds":10}` or `{"min_output_amount": "100"}`.